controllers/perconaservermysqlbackup_controller.go

```diff
@@ -45,16 +68,229 @@ type PerconaServerMySQLBackupReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
 func (r *PerconaServerMySQLBackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	l := log.FromContext(ctx).WithName("PerconaServerMySQLBackup")
+
+	rr := ctrl.Result{RequeueAfter: 5 * time.Second}
+
+	cr := &apiv1alpha1.PerconaServerMySQLBackup{}
+	if err := r.Client.Get(ctx, req.NamespacedName, cr); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return rr, nil
+		}
+
+		return rr, errors.Wrapf(err, "get %v", req.NamespacedName.String())
+	}
+
+	status := cr.Status
+
+	defer func() {
+		if status.State == cr.Status.State && status.Destination == cr.Status.Destination {
+			return
+		}
+
+		err := k8sretry.RetryOnConflict(k8sretry.DefaultRetry, func() error {
+			cr := &apiv1alpha1.PerconaServerMySQLBackup{}
+			if err := r.Client.Get(ctx, req.NamespacedName, cr); err != nil {
+				return errors.Wrapf(err, "get %v", req.NamespacedName.String())
+			}
+
+			cr.Status = status
+			l.Info("Updating status", "state", cr.Status.State)
+			if err := r.Client.Status().Update(ctx, cr); err != nil {
+				return errors.Wrapf(err, "update %v", req.NamespacedName.String())
+			}
+
+			return nil
+		})
+		if err != nil {
+			l.Error(err, "failed to update status")
+		}
+	}()
+
+	cluster := &apiv1alpha1.PerconaServerMySQL{}
+	nn := types.NamespacedName{Name: cr.Spec.ClusterName, Namespace: cr.Namespace}
+	if err := r.Client.Get(ctx, nn, cluster); err != nil {
+		return rr, errors.Wrapf(err, "get %v", nn.String())
+	}
+
+	if err := cluster.CheckNSetDefaults(r.ServerVersion); err != nil {
+		return rr, errors.Wrapf(err, "check and set defaults for %v", nn.String())
+	}
+
+	switch cr.Status.State {
+	case apiv1alpha1.BackupFailed, apiv1alpha1.BackupSucceeded:
+		// run finalizers
+		return rr, nil
+	}
+
+	if cluster.Spec.Backup == nil {
+		l.Info("spec.backup stanza not found in PerconaServerMySQL CustomResource")
+		return rr, nil
+	}
+
+	storage, ok := cluster.Spec.Backup.Storages[cr.Spec.StorageName]
+	if !ok {
+		return rr, errors.Errorf("%s not found in spec.backup.storages in PerconaServerMySQL CustomResource", cr.Spec.StorageName)
+	}
+
+	if cluster.Status.MySQL.State != apiv1alpha1.StateReady {
+		l.Info("Cluster is not ready", "cluster", cr.Name)
+		return rr, nil
+	}
+
+	j := &batchv1.Job{}
```

Comment:
`j `and `nn`  , please add more letters :)

Reply:
I changed the `j` but I'm preserving the `nn` since we're using the same in multiple places.

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/77/commits/a9ffeb86eaf8d0f58df0e91083c9855677550783